mindsdb/api/executor/sql_query/steps/map_reduce_step.py

```diff
@@ -64,10 +55,98 @@ class MapReduceStepCall(BaseStepCall):
 
     bind = MapReduceStep
 
-    def call(self, step):
+    def call(self, step: MultipleSteps):
         if step.reduce != 'union':
             raise LogicError(f'Unknown MapReduceStep type: {step.reduce}')
 
+        partition = getattr(step, 'partition', None)
+
+        if partition is not None:
+            data = self._reduce_partition(step, partition)
+
+        else:
+            data = self._reduce_vars(step)
+
+        return data
+
+    def _reduce_partition(self, step, partition):
+        if not isinstance(partition, int):
+            raise NotImplementedError('Only integers are supported in partition definition.')
+
+        input_idx = step.values.step_num
+        input_data = self.steps_data[input_idx]
+        input_columns = list(input_data.columns)
+
+        substeps = step.step
+        if not isinstance(substeps, list):
+            substeps = [substeps]
+
+        data = ResultSet()
+
+        df = input_data.get_raw_df()
+
+        # tasks
+        def split_data_f(df):
+            chunk = 0
+            while chunk * partition < len(df):
+                # create results with partition
+                df1 = df[chunk * partition: (chunk + 1) * partition]
```

Comment:
Just in case, we may wanna control for `partition > 0`, too, because if it's negative this won't behave correctly.

Reply:
Added checking value of partition